app/src/main/java/com/eveningoutpost/dexdrip/NavDrawerBuilder.java

```diff
@@ -77,18 +77,23 @@ public NavDrawerBuilder(final Context context) {
                                 && !Ob1G5CollectionService.isG5WantingCalibration()) { //Put steps in place to discourage over calibration
                                     this.nav_drawer_options.add(context.getString(R.string.override_calibration));
                                     this.nav_drawer_intents.add(new Intent(context, CalibrationOverride.class));
-                                } else {
+                                } else { // G5 or old G6 after initial calibration and long enough after previous calibration
                                     this.nav_drawer_options.add(context.getString(R.string.add_calibration));
                                     this.nav_drawer_intents.add(new Intent(context, AddCalibration.class));
                                 }
-                            } else {
+                            } else {  // G5 or old G6 not long after a calibration
                                 this.nav_drawer_options.add(context.getString(R.string.cannot_calibrate_right_now));
                                 this.nav_drawer_intents.add(new Intent(context, Home.class));
                             }
                         } else {
                             if (BgReading.isDataSuitableForDoubleCalibration() || Ob1G5CollectionService.isG5WantingInitialCalibration()) {
-                                this.nav_drawer_options.add(context.getString(R.string.initial_calibration));
-                                this.nav_drawer_intents.add(new Intent(context, DoubleCalibrationActivity.class));
+                                if (last_two_bgReadings.size() > 1) {  // Firefly after second reading, now, gets add calibration.
```

Comment:
I'm unsure whether this is the correct check, especially after `isG5WantingInitialCalibration()`. There are so many if/else here. Is there a state machine for these checks?

Reply:
Do you mean a flowchart?
How do I add a diagram to a code?
Or, do you mean something else?

Reply:
There is an OR statement between `isDataSuitableForDoubleCalibration()` and `isDataSuitableForDoubleCalibration`

I may have confused people by adding my comments to existing code.

I looked at the existing code and found where initial calibration is located.  That's where a Firefly triggers after its  third reading after a warm-up.   The only other device that ever gets to the same spot (initial calibration) is either a G5 or an old G6 in non-native mode.  Those are the only two devices that get initial calibration option under the top left menu.

However, none of those two have readings before initial calibration.
So, by adding that test, I can create a fork that sends G5 and old G6 to one branch of the fork, and Firefly to the other branch.

Reply:
>  I'm unsure whether this is the correct check, especially after isG5WantingInitialCalibration(). 

The test is not after `isG5WantingInitialCalibration()` logically.
There is an `else` on line 88.

Reply:
> Do you mean a flowchart?
> How do I add a diagram to a code?
> Or, do you mean something else?

No, I don't mean adding a diagram (i. e. as an image) to the code. I'd like to have a technical documentation which shows the Dexcom state machine and all things around it. This can be done without any graphics in the Wiki, i. e. using [mermaid](https://mermaid-js.github.io). It is just to understand the code much better and reduces the error probability.

Reply:
We can do that.  That will benefit many PRs.  
But, why are we talking about it here in this PR?

Reply:
Because it is hard to review a nested and incomplete if/then/else statement were a further condition is added and no unit test is available. It is just a general suggestion and does not influence this PR. If we would have this kind of documentation, we would see immediately that the surrounding if clause is incomplete.

Reply:
I am going to learn mermaid and put this together.  What do you mean by doing it in the wiki?

Reply:
We need some place to store the tech doc. The wiki might be a good place because it does not require pull requests.